### PR TITLE
dialects: (maths) Using `__init__()` instead of `get()`

### PR DESCRIPTION
**Pull request by ShaolunWang**
Using `__init__()` instead of `get()` for experimental dialect `maths`. This follows the previous changes on dialect. It will also make the tests for the dialect easier to write.

### CODE REVIEW

File: xdsl/dialects/experimental/math.py [hunk 1]
```diff
@@ -95,6 +93,20 @@ class Atan2Op(IRDLOperation):
     rhs: Operand = operand_def(AnyFloat)
     result: OpResult = result_def(AnyFloat)
 
+    # TODO: use __init__
```
Comment (superlopuh) [inline xdsl/dialects/experimental/math.py:96]: ?

File: xdsl/dialects/experimental/math.py [hunk 2]
```diff
@@ -95,6 +93,20 @@ class Atan2Op(IRDLOperation):
...
+    def __init__(
+        self,
+        lhs: Operation | SSAValue,
+        rhs: Operation | SSAValue,
+        fastmath: FastMathFlagsAttr | None = None,
+    ):
+        lhs = SSAValue.get(lhs)
```
Comment (superlopuh) [inline xdsl/dialects/experimental/math.py:103]: this isn't necessary `IRDLOperation.__init__` accepts `Operation | SSAValue`

**Review by superlopuh — COMMENTED** [review]
Thank you for starting this! Can you please convert the rest of the TODOs?

**Review by compor — COMMENTED** [review]
Updating the rest of `__init__`s would be great. Thanks

File: xdsl/dialects/experimental/math.py (hunk 1 above)
Comment (ShaolunWang) [reply xdsl/dialects/experimental/math.py:96]: Sorry I forgot to remove the todo after changing that

**Comment by ShaolunWang** [comment]
I just checked my pr and for some reason my script to update this wasn't working on all of them, I will fix them asap.

File: xdsl/dialects/experimental/math.py (hunk 2 above)
Comment (ShaolunWang) [reply xdsl/dialects/experimental/math.py]: Currently I'm doing     
```
return super().__init__(
            attributes=attributes,
            operands=[lhs, rhs],
            result_types=[SSAValue.get(lhs).type],
        )
```
Which is what's currently been used in gpu dialect. I'm not sure whether this is the best way.

**Comment by ShaolunWang** [comment]
> script
Technically it's a set of vim macros that runs on a single file, which can be scripted

**Comment by ShaolunWang** [comment]
Yes, I'm happy to help 👍

**Comment by webmiche** [comment]
(Or upstreaming the macros into some new folder, if that is easy enough).

File: xdsl/dialects/experimental/math.py [hunk 3]
```diff
@@ -589,19 +570,19 @@ class FmaOp(IRDLOperation):
...
         b: Operation | SSAValue,
         c: Operation | SSAValue,
         fastmath: FastMathFlagsAttr | None = None,
-    ) -> FmaOp:
+    ):
         attributes = {"fastmath": fastmath}
 
         a = SSAValue.get(a)
         b = SSAValue.get(b)
         c = SSAValue.get(c)
```
Comment (webmiche) [inline xdsl/dialects/experimental/math.py:584]: You can remove these two, right?

Comment (webmiche) [reply xdsl/dialects/experimental/math.py:584]: There are even more of those still left.

**Review by webmiche — APPROVED** [review]
Minor comments, otherwise LGTM

File: xdsl/dialects/experimental/math.py (hunk 3 above)
Comment (webmiche) [reply xdsl/dialects/experimental/math.py]: There are still 26 of these left.

**Review by compor — APPROVED** [review]

**Review by webmiche — APPROVED** [review]

**Pull request merged**